core/iwasm/aot/aot_runtime.c

```diff
@@ -1684,8 +1686,14 @@ aot_instantiate(AOTModule *module, AOTModuleInstance *parent,
     if (stack_size == 0)
         stack_size = DEFAULT_WASM_STACK_SIZE;
 #if WASM_ENABLE_SPEC_TEST != 0
+#if WASM_ENABLE_TAIL_CALL == 0
     if (stack_size < 128 * 1024)
         stack_size = 128 * 1024;
+#else
+    /* Some tail-call cases require large operand stack */
+    if (stack_size < 10 * 1024 * 1024)
```

Comment:
Tail call case already patched in this commit, should we enlarge the stack size here?

Reply:
The count of tail call for linux was patched but it is also large (~100000), it needs large operand stack. Do you suggest to decrease the size for both linux and nuttx?

Reply:
Is [gc_nuttx_tail_call.patch](https://github.com/bytecodealliance/wasm-micro-runtime/pull/2975/files#diff-6098a9fa70e8d4002392d870fe51ae3c952e347ed0235427dd4a56803fa05ea4) only for NuttX or not?

If this file is only for NuttX, maybe we can make it smaller?
If not, this file should be renamed

Reply:
It is only for nuttx. For GC spec test, we first apply patch `gc_ignore_cases.patch`, in which the count is reduced to about 100000, which is good for Ubuntu. And if we are testing nuttx qemu, then `gc_nuttx_tail_call.patch` is applied again to reduce the count to about 1000.

Reply:
So, ` if (stack_size < 10 * 1024 * 1024)` can be removed here?

Reply:
No, gc spec test in Ubuntu requires large operand stack, though the recursive count for tail call has been reduced to 100000.